File: experiments/runs/2026-05-23_11-00-44_claim-office-example-mapping_v8b-delayed-refactor-native_opus-4-7-portkey-no-thinking/src/catalogue.ts
import { Item } from "./types.js";

interface MainItemSpec {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEMS: Record<string, MainItemSpec> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

const COMPONENT_TYPES = new Set(["rune", "moonstone"]);

export const COMPONENT_INSURANCE_VALUE = 250;
export const COMPONENT_BASE_PREMIUM = 25;
export const BLOCK_SIZE = 3;
export const BLOCK_BASE_PREMIUM = 60;

export function isMainItem(type: string): boolean {
  return type in MAIN_ITEMS;
}

export function isComponent(type: string): boolean {
  return COMPONENT_TYPES.has(type);
}

export function isKnownType(type: string): boolean {
  return isMainItem(type) || isComponent(type);
}

export function assertKnownType(type: string): void {
  if (!isKnownType(type)) {
    throw new Error(`Unknown item type: ${type}`);
  }
}

export function insuranceValueOf(item: Item): number {
  if (isMainItem(item.type)) return MAIN_ITEMS[item.type].insuranceValue;
  if (isComponent(item.type)) return COMPONENT_INSURANCE_VALUE;
  throw new Error(`Unknown item type: ${item.type}`);
}

export function mainItemBasePremium(type: string): number {
  return MAIN_ITEMS[type].basePremium;
}

File: experiments/runs/2026-09-21_23-56-52_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/catalogue.py
"""The MHPCO price list."""
from fractions import Fraction

from errors import ClaimOfficeError

COMPONENT_INSURANCE_VALUE = Fraction(250)
COMPONENT_BASE_PREMIUM = Fraction(25)
BLOCK_SIZE = 3
BLOCK_BASE_PREMIUM = Fraction(60)

_MAIN_ITEMS = {
    "sword": (Fraction(1000), Fraction(100)),
    "amulet": (Fraction(600), Fraction(60)),
    "staff": (Fraction(800), Fraction(80)),
    "potion": (Fraction(400), Fraction(40)),
}

_COMPONENTS = ("rune", "moonstone")


def is_known(item_type):
    return item_type in _MAIN_ITEMS or item_type in _COMPONENTS


def is_component(item_type):
    return item_type in _COMPONENTS


def insurance_value(item_type):
    if item_type in _MAIN_ITEMS:
        return _MAIN_ITEMS[item_type][0]
    if item_type in _COMPONENTS:
        return COMPONENT_INSURANCE_VALUE
    raise ClaimOfficeError(f"unknown item type: {item_type!r}")


def base_premium(item_type):
    if item_type in _MAIN_ITEMS:
        return _MAIN_ITEMS[item_type][1]
    if item_type in _COMPONENTS:
        return COMPONENT_BASE_PREMIUM
    raise ClaimOfficeError(f"unknown item type: {item_type!r}")
